Issue #103 - Integration tests for sign up resource

omatheusmesmo opened the pull request:
Hello,

I have followed the guidelines outlined in CONTRIBUTING.md. 

Please let me know if any modifications are required. 

This pull request fixes #103.

mcruzdev commented:
Hi @omatheusmesmo could you rebase and to solve the imports?

mcruzdev commented:
The pipeline error is related to the code style. Try to execute the following goal: net.revelc.code:impsort-maven-plugin:1.12.0:sort

omatheusmesmo commented:
Thanks for the support @mcruzdev , I correct the imports order and added a test for phone number too. Could you please approve the workflows runs again?

matheusandre1 commented:
cc: @omatheusmesmo, Hello Matheus, All very well?

Well, if you'll allow me...

This error in the pipeline is related to OpenAPI, even though the test class hasn't been modified.

The simplest way is to put, I believe, this: quarkus.langchain4j.openai.enabled=false
but there must be other ways to do this to maintain the integrity of the tests, but that's up to you @mcruzdev 

We should have a standard for this, to guarantee good testing standards, since there can be various test cases, test containers, etc., an isolated database for testing, possibly containing a src/test/resources/application.properties file.

matheusandre1 commented:
@omatheusmesmo Can you rebase this?

omatheusmesmo commented:
Hi @matheusandre1, thanks for the suggestion! I added `%test.quarkus.langchain4j.openai.enable-integration=false` to disable the OpenAI integration during tests, and used a default placeholder (`${OPENAI_API_KEY:unused}`) so the config resolves without requiring the env var. The branch has been rebased onto latest main and all 10 tests pass locally.

omatheusmesmo committed:
test: add integration tests for SignUpResource with test profile configuration

- Add SignUpResourceTest with integration tests covering sign-up scenarios
- Add test profile in application.properties with disabled langchain4j OpenAI
- Update SignUpResource to use ProblemResponse and include phone number
- Reorder validation annotations in SignUpRequest

Signed-off-by: Matheus Oliveira <matheus.6148@gmail.com>
